### PR TITLE
Bug 1071938 - New admin UI created

### PR DESCRIPTION
**Pull request by ex-Dev**
New admin UI created, allowing admins to set/unset new admins. Only admin users can view the new menu item.

Includes:
A little change on README.md file.
From within test directory I had to use "." (dot) instead "test" as directory argument for the "docker build" command.

### CODE REVIEW

File: pulseguardian/web.py [hunk 1]
```diff
@@ -127,6 +130,25 @@ def decorated_function(*args, **kwargs):
...
+    specific roles
+    """
+    def requires_role_wrap(f):
+        @wraps(f)
+        def decorated_function(*args, **kwargs):
+            if ADMIN_ROLE in roles and not g.user.admin:
+                """404: Non admin user not need know that this route exists.
+                Http 401 can be returned instead."""
+                abort(404)
+
```
Comment (markrcote) [inline pulseguardian/web.py]: Drop all the blank lines here and in the rest of the function.

File: pulseguardian/web.py [hunk 2]
```diff
@@ -264,6 +295,27 @@ def delete_pulse_user(pulse_username):
...
 
+@app.route('/user/<user_id>/set-admin', methods=['PUT'])
+@requires_login
+@requires_role([ADMIN_ROLE])
+def set_user_admin(user_id):
+    if 'isAdmin' not in request.json:
+        abort(400)
+
+    try:
+        isAdmin = request.json['isAdmin']
```
Comment (markrcote) [inline pulseguardian/web.py]: This should be `is_admin` following PEP 8.

**Comment by ex-Dev** [comment]
Mark,
I think that all necessary changes was made, if I forgot something, let me know.

File: pulseguardian/static/js/users_listing.js [hunk 3]
```diff
@@ -5,4 +5,38 @@
...
+    	var userData = usersTable.row($(this).closest('tr')).data();
+    	var confirmMsg = 'Change the admin role to ' + userData[0] + ' ?';
+    	var confirmation = confirm(confirmMsg);
+
+    	if(confirmation){
+    		var isAdmin = $(this).is(':checked');
+    		var urlApi = '/user/' + $(this).data('uid') + '/set-admin'
+
+    		$.ajax({
+    			type: 'PUT',
```
Comment (markrcote) [inline pulseguardian/static/js/users_listing.js]: Actually sorry, I was thinking in Python terms here. This is probably fine. I don't have a validator file but that's an excellent idea.

Comment (ex-Dev) [inline pulseguardian/static/js/users_listing.js]: Take a look in the ShipIt repo, the releng team use a .jscs to validate the js files on Travis build.

**Comment by ex-Dev** [comment]
Hi Mark, tell me if this PR needs more changes, I keep following this thread!

File: pulseguardian/web.py [hunk 4]
```diff
@@ -129,6 +132,17 @@ def decorated_function(*args, **kwargs):
     return decorated_function
 
 
+def requires_admin(f):
+    """Decorator for views that are allowed for admin users only"""
+    @wraps(f)
+    def decorated_function(*args, **kwargs):
+        if not g.user.admin:
+            """404: Non admin user not need know that this route exists."""
```
Comment (camd) [inline pulseguardian/web.py]: The wording is a bit odd on this.  How about something like: 

    404: Non-admin user does not have access to this route.

**Comment by camd** [comment]
@ex-Dev : If you get around to it, would you also rebase this on latest master?  I'll probably do that tomorrow if I don't hear from you beforehand.

I am terribly sorry for the delay.  You've been amazingly patient with me.  And thanks for the awesome PR.  This is a cool feature!

**Comment by camd** [comment]
I needed to rebase and tweak one error message.  I kept your commit intact, however.  This was merged in PR https://github.com/mozilla/pulseguardian/pull/24

**Comment by ex-Dev** [comment]
@camd thanks for rebasing. Currently I'm working on a Balrog bug and I had no time in the last weekend to rebase this.

**Comment by camd** [comment]
No problem at all.  I’m happy to do it.  :)  I was the one dragging my feed getting it reviewed.  It’s the LEAST I could do.  :)

Thanks again for the new feature.  It’s been pushed to production this morning.
-Cam

**Pull request closed without merging**